packages/flutter/test/material/chip_test.dart

```diff
@@ -130,6 +132,66 @@ Future<void> _testConstrainedLabel(
   expect(chipSize.height, chipParentHeight);
 }
 
+Future<void> _pumpCheckmarkChip<T>(
+    WidgetTester tester, {
+      Color themeColor,
+      Color color,
+      Brightness brightness = Brightness.light
+    }) async {
+  Widget chip;
+  if (T == InputChip) {
+    chip = InputChip(
+      label: const Text('InputChip'),
+      selected: true,
+      showCheckmark: true,
+      checkmarkColor: color,
+    );
+  } else if (T == FilterChip) {
+    chip = FilterChip(
+      label: const Text('FilterChip'),
+      selected: true,
+      showCheckmark: true,
+      checkmarkColor: color,
+      onSelected: (bool _) { },
+    );
+  } else {
+    throw UnsupportedError('Chip type must be InputChip of FilterChip');
+  }
+
+  await tester.pumpWidget(
+      _wrapForChip(
+          brightness: brightness,
+          child: Builder(
+            builder: (BuildContext context) {
+              final ChipThemeData chipTheme = ChipTheme.of(context);
+              return ChipTheme(
+                data: themeColor == null ? chipTheme : chipTheme.copyWith(
+                  checkmarkColor: themeColor,
+                ),
+                child: chip,
+              );
+            },
+          )
+      )
+  );
+}
+
+void _expectCheckmarkColor(Finder finder, Color color, { bool isDark = false }) {
```

Comment:
This function actually does more than check the checkmark color.

Consider making two functions.

```dart
_expectCheckmarkColor(Finder finer, color)
// and
_expectChipOverlay(Finder finder, { bool isDark = false });
```

And then call those both explicitly in each test.

Reply:
It checks the overlay first because the expect paints API has to cover all the paths that are painted in order. The check mark is the second path to be painted, so we have to add the first path. However, I was able to remove the color from the first path, since that is not what we care about, and it allowed for isDark to be removed.